### PR TITLE
Clean up null assumptions for project.dart dependencies

### PR DESCRIPTION
**Pull request by jmagman**
I started migrating `project.dart` and all its dependencies, and the diff was huge.
Pull out the cleanup parts in project, flutter_plugins, and cmake (kind of a random grouping)  that can be done without actually migrating to null safety.

Part of #71511

### CODE REVIEW

File: packages/flutter_tools/lib/src/project.dart [hunk 1]
```diff
@@ -334,10 +350,14 @@ class FlutterProject {
 
   /// Returns a json encoded string containing the [appName], [version], and [buildNumber] that is used to generate version.json
   String getVersionInfo()  {
+    final String buildName = manifest.buildName;
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/project.dart:353]: This method is mostly used to generate dep files.

File: packages/flutter_tools/lib/src/project.dart [hunk 2]
```diff
@@ -613,25 +633,33 @@ class IosProject extends FlutterProjectPlatform implements XcodeBasedProject {
       scheme,
     );
     final XcodeProjectBuildContext buildContext = XcodeProjectBuildContext(environmentType: environmentType, scheme: scheme, configuration: configuration);
-    return _buildSettingsByBuildContext[buildContext] ??= await _xcodeProjectBuildSettings(buildContext);
+    if (_buildSettingsByBuildContext[buildContext] == null) {
+      final Map<String, String> calculatedBuildSettings = await _xcodeProjectBuildSettings(buildContext);
+      if (calculatedBuildSettings != null) {
+        _buildSettingsByBuildContext[buildContext] = calculatedBuildSettings;
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/project.dart:639]: Avoid setting a _buildSettingsByBuildContext value to `null`.

File: packages/flutter_tools/lib/src/project.dart [hunk 3]
```diff
@@ -975,12 +1004,13 @@ to migrate your project.
       logger: globals.logger,
       templateRenderer: globals.templateRenderer,
     );
+    final String androidIdentifier = parent.manifest.androidPackage ?? 'com.example.${parent.manifest.appName}';
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/project.dart:1007]: This seems like a good fallback, already used in https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/packages/flutter_tools/lib/src/android/deferred_components_prebuild_validator.dart#L271

File: packages/flutter_tools/lib/src/project.dart [hunk 4]
```diff
@@ -791,12 +819,13 @@ class IosProject extends FlutterProjectPlatform implements XcodeBasedProject {
       logger: globals.logger,
       templateRenderer: globals.templateRenderer,
     );
+    final String iosBundleIdentifier = parent.manifest.iosBundleIdentifier ?? 'com.example.${parent.manifest.appName}';
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/project.dart:822]: See `androidIdentifier` comment, `parent.manifest.appName` seems like a fine fallback value.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**